File: aerox.Runtime.Widgets/Events/ResizeEvent.cs
using aerox.Runtime.Math;
using aerox.Runtime.Widgets.Graphics;

namespace aerox.Runtime.Widgets.Events;

public class ResizeEvent : Event
{
    public Vector2<int> Size;

    public ResizeEvent(Surface surface, Vector2<int> newSize) : base(surface)
    {
        Size = newSize;
    }
}

File: modules/widgets/lib/aerox/widgets/event/ResizeEvent.cpp
#include "aerox/widgets/event/ResizeEvent.hpp"

namespace aerox::widgets
{
    ResizeEvent::ResizeEvent(const Shared<Surface>& inSurface, const Vec2<float>& inSize) : Event(inSurface), size(inSize)
    {
        
    }
}
